### ISSUE TITLE
druid-shell no clear causes flickers on windows

### ISSUE THREAD

**Issue by Kethku** [xi-editor/druid#375]
If you use druid-shell without clearing on each paint event, you get a weird flickering effect which I assume is a result of the DXGI_SWAP_EFFECT_FLIP_SEQUENTIAL presentation strategy.

This wasn't obvious at first, and should probably be documented somewhere.

**Comment by raphlinus** [comment]
Yes, the client is responsible for validating the entire window every frame, and I agree this could be documented. However, this *definitely* is going to change as we implement incremental presentation, which is one reason we haven't been investing too much into documentation.

**Comment by Finnerale** [comment]
Is this still an issue or was it resolved with partial invalidation?
